providers/summary/clearlydefined.js

```diff
@@ -11,7 +11,10 @@ const {
   isDeclaredLicense,
   isLicenseFile,
   updateSourceLocation,
-  mergeDefinitions
+  mergeDefinitions,
+  buildUrls,
+  buildRegistryUrl,
+  buildVersionUrl
 } = require('../../lib/utils')
```

Comment:
I think we should probably move `buildUrls` out of utils now that it won't handle all coordinates thrown at it.
Probably better just to put each in it's respective spot here.

so for example `addNpmData` will add the urls for npms

Reply:
I'm not 100% sure about this.
Most of the providers use the same path to retrieve the URLs.
Only in some specific cases (Pypi, Cocoapods) we must use different ways.

Removing that function would potentially cause some code duplication.

Thoughts?

Reply:
I don't think code duplication will be a problem here

It would be slotting these into each of the methods instead of living in a big partial switch

```js
function buildRegistryUrl(spec) {
  switch (spec.provider) {
    case 'github':
      return `https://github.com/${spec.namespace}/${spec.name}`
    case 'npmjs':
      return `https://npmjs.com/package/${spec.namespace ? spec.namespace + '/' + spec.name : spec.name}`
    case 'nuget':
      return `https://nuget.org/packages/${spec.name}`
    case 'cratesio':
      return `https://crates.io/crates/${spec.name}`
    case 'mavencentral':
      return `https://mvnrepository.com/artifact/${spec.namespace}/${spec.name}`
    case 'pypi':
      return `https://pypi.org/project/${spec.name}`
    case 'rubygems':
      return `https://rubygems.org/gems/${spec.name}`
    case 'cocoapods':
      return `https://cocoapods.org/pods/${spec.name}`
    default:
      return null
  }
```

Reply:
Right, will address this soon.

Reply:
Maybe i'm wrong but i can't found the specific method for `github` in the summarizer. Is that right?

Reply:
you're right. We'll have to add one - good catch